README.md

```diff
@@ -1415,6 +1290,71 @@ To run Terragrunt locally, use the `go run` command:
 go run main.go plan
 ```
 
+#### Dependencies
+
+* Terragrunt uses `glide`, a vendor package management tool for golang. See the glide repo for
+  [installation instructions](https://github.com/Masterminds/glide#install).
+
+#### AWS credentials
+
+Terragrunt uses the official [AWS SDK for Go](https://aws.amazon.com/sdk-for-go/), which
+means that it will automatically load credentials using the 
+[AWS standard approach](https://aws.amazon.com/blogs/security/a-new-and-standardized-way-to-manage-credentials-in-the-aws-sdks/). See the Go SDK guide on [Specifying Credentials](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html) for details, but basically your options include:
+
+1. Environment Variables
+    1. `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` for credentials (and also `AWS_SESSION_TOKEN` if using [STS temporary credentials](http://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html))
+   1. `AWS_PROFILE` for the profile
+1. [AWS credentials file](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#shared-credentials-file). You can set this up by running `aws configure` and filling in the details.
+1. Run Terragrunt on an EC2 instance with an IAM Role.
+
```

Comment:
I propose that we update the IAM policy to be: 

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "AllowAllDynamoDBActionsOnAllTerragruntTables",
            "Effect": "Allow",
            "Action": "dynamodb:*",
            "Resource": [
                "arn:aws:dynamodb:*:260374580803:table/terragrunt*"
            ]
        },
        {
            "Sid": "AllowAllS3ActionsOnTerragruntBuckets",
            "Effect": "Allow",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::terragrunt*",
                "arn:aws:s3:::terragrunt*/*"
            ]
        }
    ]
}
```

This will be easy to explain to users that they can change the `terragrunt` to whatever their own naming convention is. However, in order to use the IAM policy above as written, we will need to update a Terragrunt test that does not follow this pattern. In [dynamo_lock_table_test.go](https://github.com/gruntwork-io/terragrunt/blob/7ed21eea52bfe085c67c52c9e58588786b59e769/dynamodb/dynamo_lock_table_test.go#L51), we will have to change `tableName := "table-does-not-exist"` to `tableName := "terragrunt-table-does-not-exist"` for the tests to pass. Otherwise, we get an `AccessDeniedException` from AWS when we try to describe the table `table-does-not-exist`. We could, of course, update the IAM policy, but I think that would be unnecessarily complicated and updating the test table name to follow the pattern makes more sense.

Thoughts?